### PR TITLE
feat: add get_dataset_versions method to RegisteredModelVersion object

### PR DESCRIPTION
**Pull request by baasitsh-verta**
## Impact and Context

Allows fetching all dataset versions associated with a RegisteredModelVersion

## Testing
- [x] Unit test
```
(base) baasitsharief@Baasits-MBP tests % pytest registry/model_version/test_log_dataset.py 
=================================================================== test session starts ====================================================================
platform darwin -- Python 3.9.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/baasitsharief/code/VertaAI/modeldb/client/verta/tests, configfile: pytest.ini
plugins: anyio-3.5.0, hypothesis-6.75.2, dash-2.9.1
collected 1 item                                                                                                                                           

registry/model_version/test_log_dataset.py .                                                                                                         [100%]

==================================================================== 1 passed in 4.83s =====================================================================
```
- [ ] Deployed to dev env
- [ ] Other (explain) 

## Reverting
- [ ] Contains Migration - _Do Not Revert_

### CODE REVIEW

File: client/verta/verta/registry/entities/_modelversion.py [hunk 1]
```diff
@@ -1665,6 +1665,31 @@ def get_dataset_version(self, key):
...
+        for dataset in self._msg.datasets:
+            dataset_versions[dataset.key] = _dataset_version.DatasetVersion(
+                    self._conn,
+                    self._conf,
+                    _dataset_version.DatasetVersion._get_proto_by_id(
+                        self._conn, dataset.linked_artifact_id
+                    ),
+                )
+        
+        return dataset_versions
```
Comment (ewagner-verta) [inline client/verta/verta/registry/entities/_modelversion.py]:
```suggestion
    def get_dataset_versions(self):
        """
        Gets all DatasetVersions associated with this Model Version.

        Returns
        -------
        list of :class:`~verta.dataset.entities.DatasetVersion`
            DatasetVersions associated with this Model Version.
        """
        self._refresh_cache()

        dataset_versions = list()

        for dataset in self._msg.datasets:
            dataset_versions.append(
                _dataset_version.DatasetVersion(
                    self._conn,
                    self._conf,
                    _dataset_version.DatasetVersion._get_proto_by_id(
                        self._conn, dataset.linked_artifact_id
                    ),
                )
            )
        return dataset_versions
```

Comment (ewagner-verta) [reply client/verta/verta/registry/entities/_modelversion.py]: Screen Shot 2023-05-10 at 11 39 42 AM
^^ That renders a bit oddly with the \s, so I simplified it to:
Screen Shot 2023-05-10 at 12 12 02 PM


Also, the general design pattern in other client functions that `get_some_things()` typically return a list of the requested objects, so this suggested change converts the type accordingly.

Lastly, we use [black](https://pypi.org/project/black/) for auto-formatting.  You may want to install and use it going forward (some spacing issues), plus it works with Jupiter notebooks which is convenient.

File: client/verta/tests/registry/model_version/test_log_dataset.py [hunk 2]
```diff
@@ -32,6 +32,11 @@ def test_log_dataset(self, client, model_version, dataset):
         dataset = model_version.get_dataset_version(key3)
         assert dataset_version3.id == dataset.id
 
+        dataset_versions =  model_version.get_dataset_versions()
+        assert dataset_versions[key1].id == dataset_version1.id
+        assert dataset_versions[key2].id == dataset_version2.id
+        assert dataset_versions[key3].id == dataset_version3.id
+
```
Comment (ewagner-verta) [inline client/verta/tests/registry/model_version/test_log_dataset.py]:
```suggestion
        dataset_versions = model_version.get_dataset_versions()
        assert dataset_versions[0].id == dataset_version1.id
        assert dataset_versions[1].id == dataset_version2.id
        assert dataset_versions[2].id == dataset_version3.id

```

Comment (ewagner-verta) [reply client/verta/tests/registry/model_version/test_log_dataset.py]: Removes extra space (see comment on auto-formatting below) and updates assertions for type change recommended below.

**Review by ewagner-verta — CHANGES_REQUESTED** [review]

File: client/verta/verta/registry/entities/_modelversion.py (hunk 1 above)
Comment (baasitsh-verta) [reply client/verta/verta/registry/entities/_modelversion.py]: Updated and installed black 👍

**Review by ewagner-verta — APPROVED** [review]

**Pull request merged**